### DISCUSSION TITLE
getInfo() does not work

### DISCUSSION THREAD

**Discussion by DennisSkov** [opengsq/opengsq-php#3] [General]
I've tried all the possible combinations that I can think of but I haven't gotten the `getInfo()` function working once. It always throws an exception because the IP and post does not exist.

This is one of the examples I'm using:
```
<?php

// Include the Composer autoloader
require_once 'vendor/autoload.php';

// Import the Vcmp class from the OpenGSQ\Protocols namespace
use OpenGSQ\Protocols\EOS;

$clientId = 'xyza78916PZ5DF0fAahu4tnrKKyFpqRE';
$clientSecret = 'j0NapLEPm3R3EOrlQiM8cRLKq3Rt02ZVVwT0SkZstSg';
$deploymentId = '0a18471f93d448e2a1f60e47e03d3413';
$grantType = 'external_auth';
$externalAuthType = 'deviceid_access_token';

try {
    $externalAuthToken = EOS::getExternalAuthToken($clientId, $clientSecret, $externalAuthType);
    $accessToken = EOS::getAccessToken($clientId, $clientSecret, $deploymentId, $grantType, $externalAuthType, $externalAuthToken);
    $matchmaking = EOS::getMatchmaking($deploymentId, $accessToken);
    $eos = new EOS('34.22.86.173', '8211', $deploymentId, $accessToken);
    echo '<pre>';
    print_r($matchmaking);
    print_r($eos->getInfo());
} catch (Exception $e) {
    echo '<pre>';
    print_r($e);
}
```
The IP and port from the example is just a random server from the $matchmaking array.

Any idea why it does not work? Am I doing something wrong?

**Comment by BattlefieldDuck** [comment]
The `getInfo()` function doesn't work for Palworld. Use this instead, 

```php
$ipAddress = '34.22.86.173';
$port = 8211;

$filter = array(
    'criteria' => array(
        array(
            'key' => 'attributes.ADDRESS_s',
            'op' => 'EQUAL',
            'value' => $ipAddress
        ),
        array(
            'key' => 'attributes.GAMESERVER_PORT_l',
            'op' => 'EQUAL',
            'value' => $port
        )
    )
);

$matchmaking = EOS::getMatchmaking($deploymentId, $accessToken, $filter);
var_dump($matchmaking)
```

**Reply by DennisSkov** [reply]
Perhaps we should be able to pass `$filter` to the `getInfo()` function?
The provided workaround doesn't actually solve the issue because the `getInfo()` function has a hardcoded value for `$filter`.

**Reply by BattlefieldDuck** [reply]
We should always use `$matchmaking = EOS::getMatchmaking($deploymentId, $accessToken, $filter);` since it is more flexible on the `$filter` settings. It is not a must to use `getInfo` function.

**Reply by DennisSkov** [reply]
Gotcha! I completely missed that we could just use the `getMatchmaking()` function. That makes a lot more sense.
